src/lib/io/sol-iio.c

```diff
@@ -602,6 +602,7 @@ channel_get_pure_name(struct sol_iio_channel *channel)
             streq(channel_name_suffix,  "_y") || streq(channel_name_suffix,  "_z")) {
 
             channel_pure_name = strndup(channel_name, channel_name_len - 2);
+            SOL_NULL_CHECK(channel_pure_name, NULL);
```

Comment:
:+1:  for that. Please since you are touching this function, there is a `calloc` here that should be checked too.